### PR TITLE
switch if length of data set rather than data

### LINKED ISSUE #369: Plot axis bounds for repeating values
The [calculate_axis_bounds](https://github.com/force-h2020/force-wfmanager/blob/9f070c8b7ec279b89dac6f78c1aec717ae3f243e/force_wfmanager/ui/review/plot.py#L359) method does not properly handle the case when the `len(data) > 1` but the actual data values are the same. 

Consider the case `data = [1,1,1]`: the bounds must be calculated as
```
            bounds = (data[0] - 0.5, data[0] + 0.5)
```
While currently the first branch `if len(data) > 1:` is executed.

To fix this bug, the `len(set(data))` should be used instead of `len(data)`.

### PR DESCRIPTION
**Pull request by sparsonslab**
Closes #369. Length of set is calculated before if-else ladder, as otherwise set would have to be called twice on a long list with a set size of 1.

### CODE REVIEW

**Comment by flongford** [comment]
> ... excuse me forgot the commit comment. Now we don't need to use the set operation. Just check the equality of the first two elements (if len > 1)

I'm not sure this would still work if other elements in the list had the possessed the same value. Is there any preference for not using `set` here?

**Comment by sparsonslab** [comment]
> > ... excuse me forgot the commit comment. Now we don't need to use the set operation. Just check the equality of the first two elements (if len > 1)
> 
> I'm not sure this would still work if other elements in the list had the possessed the same value. Is there any preference for not using `set` here?

Because set() might take a long time on a large list. I know these new-fangled computers are very fast, but its still neat to act like we need to be efficient. 
note, this wouldn't make any difference if len() was o(n) as well, but it is actually o(1) as it just returns an attribute of the list.

**Comment by flongford** [comment]
> > > ... excuse me forgot the commit comment. Now we don't need to use the set operation. Just check the equality of the first two elements (if len > 1)
> > 
> > 
> > I'm not sure this would still work if other elements in the list had the possessed the same value. Is there any preference for not using `set` here?
> 
> I think I see.

Now I'm questioning my logic - hang on, you might have been right previously....

**Comment by sparsonslab** [comment]
> > > > ... excuse me forgot the commit comment. Now we don't need to use the set operation. Just check the equality of the first two elements (if len > 1)
> > > 
> > > 
> > > I'm not sure this would still work if other elements in the list had the possessed the same value. Is there any preference for not using `set` here?
> > 
> > 
> > I think I see.
> 
> Now I'm questioning my logic - hang on, you might have been right previously....

No, you're right. My code would have failed with e.g. data = [1, 1, 0]

**Comment by flongford** [comment]
> > > > > ... excuse me forgot the commit comment. Now we don't need to use the set operation. Just check the equality of the first two elements (if len > 1)
> > > > 
> > > > 
> > > > I'm not sure this would still work if other elements in the list had the possessed the same value. Is there any preference for not using `set` here?
> > > 
> > > 
> > > I think I see.
> > 
> > 
> > Now I'm questioning my logic - hang on, you might have been right previously....
> 
> No, you're right. My code would have failed with e.g. data = [1, 1, 0]

Right, yes that makes sense - phew, got there in the end

File: force_wfmanager/ui/review/tests/test_plot.py [hunk 1]
```diff
@@ -230,7 +230,7 @@ def test_select_plot_axis(self):
             mock_update.assert_called()
 
         self.plot._update_plot()
-        self.assertEqual((1.8, 4.2, 3.0, 3.0), self.plot._get_plot_range())
+        self.assertEqual((1.8, 4.2, 2.5, 3.5), self.plot._get_plot_range())
```
Comment (flongford) [inline force_wfmanager/ui/review/tests/test_plot.py:233]: Just checking - presumably this test was incorrect before? 

It looks like the updated version should be what we expect to return

**Comment by flongford** [comment]
Ah, one other addition that we should include is a unit test of `calculate_axis_bounds ` for the edge case where `data = [1, 1, 1]` (to show the bug is fixed)

File: force_wfmanager/ui/review/tests/test_plot.py (hunk 1 above)
Comment (sparsonslab) [reply force_wfmanager/ui/review/tests/test_plot.py:233]: if you do print(data) at the start of calculate_axis_bounds(data), you'll find
[2. 4.]   > 1.8, 4.2
[3. 3.]  > 2.5, 3.5  (not 3.0, 3.0)

So yes the original assert was wrong.

Comment (flongford) [reply force_wfmanager/ui/review/tests/test_plot.py:233]: I agree - looking closer I think this change also provides the unit test that I requested
